src/util/strencodings.h

```diff
@@ -121,28 +121,25 @@ NODISCARD bool ParseUInt64(const std::string& str, uint64_t *out);
 NODISCARD bool ParseDouble(const std::string& str, double *out);
 
 template<typename T>
-std::string HexStr(const T itbegin, const T itend, bool fSpaces=false)
+std::string HexStr(const T itbegin, const T itend)
 {
     std::string rv;
     static const char hexmap[16] = { '0', '1', '2', '3', '4', '5', '6', '7',
                                      '8', '9', 'a', 'b', 'c', 'd', 'e', 'f' };
-    rv.reserve((itend-itbegin)*3);
+    rv.reserve(std::distance(itbegin, itend) * 2);
```

Comment:
`#include <iterator>`?

Reply:
Thanks for noticing. Done.

Reply:
Why in `uint256.h`?

Reply:
I have no idea. Mix-up on my part. Fixed now.